### ISSUE TITLE
Support For Private Repos?

### ISSUE THREAD

**Issue by dodekeract** [r24y/github-issues#38]
Would be awesome to have that, since most of the code I currently write is private. 😢

**Comment by mdxprograms** [comment]
I second that. Would like to see that as well.

**Comment by r24y** [comment]
Planning on giving this package some (sorely-needed) love over the holidays; thank you both for the heads-up!

**Comment by mdxprograms** [comment]
Sounds great. Don't work too much over the holidays 😁

**Comment by r24y** [comment]
@mdxprograms looks like I took your advice too literally; good news though, `v0.8.0` is out and you should be able to use it with private repos once you get your key set up. Let me know if you have any issues with it.

**Comment by dodekeract** [comment]
@r24y How do I activate the private repo support? Preference panel only shows the README for me.

**Comment by r24y** [comment]
Would you mind pasting here the version of Atom and the version of `github-issues` you've got installed? You should see a "User Token" field in the preferences pane, and the "GitHub Issues: Open" command should work for public repos without configuration.  I'll reopen this issue while this remains unsolved.

**Comment by dodekeract** [comment]
http://i.imgur.com/0f70DRI.png

**Comment by pjatx** [comment]
I have my key setup and it's still not pulling in any issues.

![image](https://cloud.githubusercontent.com/assets/5004634/25355386/0f49a01c-28fc-11e7-80bd-615674e35e9c.png)

![image](https://cloud.githubusercontent.com/assets/5004634/25355394/1946e89a-28fc-11e7-8f6c-983c1653b34d.png)

Atom v1.16.0
Github-Issues 0.8.3

**Comment by dodekeract** [comment]
Since Atom 0.18 added GitHub integration, I revisited this package. I still have 0.8.3 installed, but now I can see the token field.

I created a token with "full private repo" access, but as far as I could tell from the network inspector this package hasn't requested the GitHub API. It just loads forever, without logging any sort of error in the console.

**Comment by r24y** [comment]
@dodekeract the package uses the `request-promise` package to perform network requests, and Atom's devtools only show requests made from WebKit, so they wouldn't show up anyways.

Do you have more than one repository open in Atom? I just realized that it only displays issues for one repo at a time, and if the first repo is not a Github repo then it will not attempt to load them.

**Comment by FlorianWendelborn** [comment]
@r24y No, I assumed this might be an issue and closed all other projects. I also updated to `quick-issues`, but that doesn't fix anything. :/

**Comment by ryaninvents** [comment]
Hm, this is turning into a very interesting issue 🙃 

Is there anything else that might be "special" about the setup? Maybe:
- Proxy/firewall? (I'm behind a proxy myself and it's working, but perhaps there's an edge case I'm not experiencing.)
- Permissions granted to the API key are not broad enough? (If the key did not have permission to read issues it might look like this.)
- Issues filter "Open/Closed/All" is not set correctly (e.g. filter set to "Open" but all issues are closed)? (I don't think this is the problem, but someone else just filed a bug that looked like this.)

Any further information you can provide would be helpful, since right now there aren't enough clues as to what's happening (er, I suppose, not happening).

**Comment by FlorianWendelborn** [comment]
@r24y I have this issue on both my Macs. One is a 2017 MacBook Pro and the other one is a Hackintosh. It might be macOS only.

Proxy/firewall: nothing crazy, just ordinary stuff.

Permissions granted: I granted the full `repo` permission. It's a shame that `issues` isn't a separate permission...

Filter: I have it set to `all`.

*Note: It now shows `No GitHub repos found`. Perhaps it's your GitHub detection? Can you verify that it works with repos cloned via e.g. `git clone git@github.com:dodekeract/config`?*

~~Oh, there might be something special. All my repos are just symbolic links. However, if you're not explicitly trying to disallow them they should just work. (I clone to `~/Code/*` and symlink projects to `~/.atom/projects/*`)~~ (tested the real folder, doesn't matter)

Also, my personal access token shows "never used", so this package doesn't call the API.

**Comment by ryaninvents** [comment]
Wonderful, this is very helpful. Thanks so much. I'll see what this turns up.
